src/cpp/fastdds/xtypes/serializers/idl/dynamic_type_idl.cpp

```diff
@@ -0,0 +1,1086 @@
+        const DynamicType::_ref_type& dyn_type,
+        std::ostream& idl) noexcept
+{
+    assert(dyn_type->get_kind() == TK_MAP);
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = dyn_type->get_descriptor(type_descriptor);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    idl << "map<";
+
+    const auto key_type = type_descriptor->key_element_type();
+    ret = type_kind_to_idl(key_type, idl);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    idl << ", ";
+
+    const auto value_type = type_descriptor->element_type();
+    ret = type_kind_to_idl(value_type, idl);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    BoundSeq bounds;
+    ret = container_size(dyn_type, bounds);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    if (bounds.size() == 1)
+    {
+        idl << ", " << std::to_string(bounds[0]);
+    }
+
+    idl << ">";
+
+    return ret;
+}
+
+ReturnCode_t sequence_kind_to_idl(
+        const DynamicType::_ref_type& dyn_type,
+        std::ostream& idl) noexcept
+{
+    assert(dyn_type->get_kind() == TK_SEQUENCE);
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    DynamicType::_ref_type internal_type;
+    ret = container_internal_type(dyn_type, internal_type);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    idl << "sequence<";
+
+    ret = type_kind_to_idl(internal_type, idl);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    BoundSeq bounds;
+    ret = container_size(dyn_type, bounds);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    if (1 == bounds.size())
+    {
+        idl << ", " << std::to_string(bounds[0]);
+    }
+
+    idl << ">";
+
+    return ret;
+}
+
+ReturnCode_t string_kind_to_idl(
+        const DynamicType::_ref_type& dyn_type,
+        std::ostream& idl) noexcept
+{
+    assert(dyn_type->get_kind() == TK_STRING8 || dyn_type->get_kind() == TK_STRING16);
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    if (dyn_type->get_kind() == TK_STRING16)
+    {
+        idl << "wstring";
+    }
+    else
+    {
+        idl << "string";
+    }
+
+    BoundSeq bounds;
+    ret = container_size(dyn_type, bounds);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    if (1 == bounds.size())
+    {
+        idl << "<" << std::to_string(bounds[0]) << ">";
+    }
+
+    return ret;
+}
+
+ReturnCode_t container_internal_type(
+        const DynamicType::_ref_type& dyn_type,
+        DynamicType::_ref_type& internal_type) noexcept
+{
+    ReturnCode_t ret = RETCODE_OK;
+
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = dyn_type->get_descriptor(type_descriptor);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    internal_type = type_descriptor->element_type();
+
+    return ret;
+}
+
+ReturnCode_t container_size(
```

Comment:
NIT: since this now also applies to other types other than containers you might rename it to `get_bounds` or similar.